filter/filter.go

```diff
@@ -0,0 +1,150 @@
+package filter
+
+import (
+	"bufio"
+	"context"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	utils "github.com/bolt-observer/go_common/utils"
+	"github.com/fsnotify/fsnotify"
+	"github.com/golang/glog"
+)
+
+type FileFilter struct {
+	Filter
+	WhitelistFilePath string
+	Mutex             sync.Mutex
+}
+
+func (f *FileFilter) Reload() error {
+	f.Mutex.Lock()
+	defer f.Mutex.Unlock()
+
+	r := regexp.MustCompile(`\s*#.*$`)
+
+	readFile, err := os.Open(f.WhitelistFilePath)
+
+	if err != nil {
+		return err
+	}
+
+	defer readFile.Close()
+
+	f.nodeIdWhitelist = make(map[string]struct{})
+	f.chanIdWhitelist = make(map[uint64]struct{})
+
+	fileScanner := bufio.NewScanner(readFile)
```

Comment:
we could simplify this reload() function by splitting to `reading file` and `parsing channels` parts.

Reply:
Will address that later, but good point.